x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_logic.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+import queryString from 'query-string';
+
+import { KibanaLogic } from '../../../shared/kibana';
+import { HttpLogic } from '../../../shared/http';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { EngineLogic } from '../engine';
+
+import { AnalyticsData, QueryDetails } from './types';
+
+interface AnalyticsValues extends AnalyticsData, QueryDetails {
+  dataLoading: boolean;
+}
+
+interface AnalyticsActions {
+  onAnalyticsUnavailable(): void;
+  onAnalyticsDataLoad(data: AnalyticsData): AnalyticsData;
+  onQueryDataLoad(data: QueryDetails): QueryDetails;
+  loadAnalyticsData(): void;
+  loadQueryData(query: string): string;
+}
+
+export const AnalyticsLogic = kea<MakeLogicType<AnalyticsValues, AnalyticsActions>>({
+  path: ['enterprise_search', 'app_search', 'analytics_logic'],
+  actions: () => ({
+    onAnalyticsUnavailable: true,
+    onAnalyticsDataLoad: (data) => data,
+    onQueryDataLoad: (data) => data,
+    loadAnalyticsData: true,
+    loadQueryData: (query) => query,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadAnalyticsData: () => true,
+        loadQueryData: () => true,
+        onAnalyticsUnavailable: () => false,
+        onAnalyticsDataLoad: () => false,
+        onQueryDataLoad: () => false,
+      },
+    ],
+    analyticsUnavailable: [
+      false,
+      {
+        onAnalyticsUnavailable: () => true,
+        onAnalyticsDataLoad: () => false,
+        onQueryDataLoad: () => false,
+      },
+    ],
+  }),
+  listeners: ({ actions }) => ({
+    loadAnalyticsData: async () => {
+      const { history } = KibanaLogic.values;
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const { start, end, tag } = queryString.parse(history.location.search);
+        const query = { start, end, tag, size: 20 };
+        const url = `/api/app_search/engines/${engineName}/analytics/queries`;
```

Comment:
Do we not have a `routePaths` or similar setup in Kibana?

Reply:
We don't, there's some complexity there because it would need to live in the `common/` folder to be shared between both our public and server folders, and so far I haven't been to bothered / seen a real need to DRY them out.

 I'd definitely need to be convinced of the benefit there honestly, I find it more useful to have the explicit URL immediately available so I can check my devtools/XHR calls for debugging etc. and not have to hunt down whatever URL var the source is from (I'll admit this annoys me a decent amount in `ent-search` especially since our API routes are all programmatically generated so I have to go to a 404 page to even find our API routes).